### PR TITLE
Add Extent.getHighestYAt

### LINKED ISSUE #1375: Get Highest Block at Specific Location
In Bukkit there was a `World.getHighestBlockAt(Location)` and `World.getHighestBlockYAt(Location)` method, which returned the highest non air block at the given location and the Y position of that block respectively. 

Unless there is some other really easy way to do this, I think these methods should be added to API

### PR DESCRIPTION
**Pull request by Maxqia**
**API** | [Common](https://github.com/SpongePowered/SpongeCommon/pull/1015)
This replaces #1381 and fixes #1375.

### CODE REVIEW

File: src/main/java/org/spongepowered/api/world/World.java [hunk 1]
```diff
@@ -329,6 +329,25 @@ default void setSerializationBehavior(SerializationBehavior behavior) {
...
+    int getHighestYAt(int x, int z);
+
+    /**
+     * Get the {@link Location} of the highest block that sunlight can reach in the given column.
+     * <p>This method ignores all transparent blocks, providing the highest opaque block.</p>
+     *
+     * @param position The column position
+     * @return The highest opaque block
+     */
+    default Location<World> getHighestBlockAt(Vector3i position) {
```
Comment (kashike) [inline src/main/java/org/spongepowered/api/world/World.java]: Why `Vector3i` and not `Vector2i`?

**Review by kashike — CHANGES_REQUESTED** [review]

File: src/main/java/org/spongepowered/api/world/World.java (hunk 1 above)
Comment (Maxqia) [reply src/main/java/org/spongepowered/api/world/World.java]: "Well it makes it easier to do something like this world.getHighestBlockAt(location.getBlockPosition()). "

Comment (JBYoshi) [reply src/main/java/org/spongepowered/api/world/World.java]: Maybe you could add a method for both?

Comment (Maxqia) [reply src/main/java/org/spongepowered/api/world/World.java]: That'd be pointless, nothing with locations uses Vector2i

File: src/main/java/org/spongepowered/api/world/Location.java [hunk 2]
```diff
@@ -680,6 +694,15 @@ public boolean spawnEntities(Iterable<? extends Entity> entities, Cause cause) {
         return this.getExtent().spawnEntities(entities, cause);
     }
 
+    /**
+     * Gets the highest Location at this Location
+     * @return the highest Location at this Location
```
Comment (gabizou) [inline src/main/java/org/spongepowered/api/world/Location.java:699]: New line between description and `@return`

File: src/main/java/org/spongepowered/api/world/extent/Extent.java [hunk 3]
```diff
@@ -106,6 +106,25 @@
     }
 
     /**
+     * Get the y value of the highest block that sunlight can reach in the given column.
```
Comment (gabizou) [inline src/main/java/org/spongepowered/api/world/extent/Extent.java:109]: Line length is longer than 80 characters. Please run `gradle checkstyle` to fix your warnings.

**Review by gabizou — CHANGES_REQUESTED** [review]
Checkstyle issues. Documentation style issues.

**Comment by kashike** [comment]
@Maxqia Able to get this updated?

**Pull request merged**